services/src/modules/resource-repository/types.ts

```diff
@@ -62,11 +63,17 @@ export interface UpstreamClientCredentials extends Resource {
 export interface PolicyDefinition extends Resource {
   type: PolicyType;
   code: string;
-  args?: PolicyArgsObject;
+  args?: PolicyArgsDefinitions;
   query?: PolicyQuery;
   shouldOverrideBasePolicy?: boolean;
 }
 
+export interface PolicyArgDefinition {
+  type: string;
+  default?: string;
+  optional?: boolean;
```

Comment:
I think we can make this field internal. It can be set in registry by calling `parseType` and `isNotNullType` functions from `graphql`. So we can validate that user provided valid type and decide if the argument is required or optional according to the type itself